.buildkite/hooks/pre-command

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+source .buildkite/scripts/common.sh
+
+set -euo pipefail
+
+# go_mod_changeset=(
+#   "^go.mod"
+#   )
+
+docs_changeset=(
+  ".*\\.(asciidoc|md)"
+  "deploy/kubernetes/.*-kubernetes\\.yaml"
+  )
+
+# packaging_changeset=(
+#   "^dev-tools/packaging/.*"
+#   ".go-version"
+#   )
+
+if ! are_changed_only_paths "${docs_changeset[@]}" ; then
+  ONLY_DOCS="false"
+  echo "Changes include files outside the docs_changeset vairiabe. ONLY_DOCS=$ONLY_DOCS."
+else
+  echo "All changes are related to DOCS. ONLY_DOCS=$ONLY_DOCS."
+fi
+
+# if are_paths_changed "${go_mod_changeset[@]}" ; then
+#   GO_MOD_CHANGES="true"
+# fi
+
+# if are_paths_changed "${packaging_changeset[@]}" ; then
+#   PACKAGING_CHANGES="true"
+# fi
+
+# Define step conditions
+
+if [[ "${BUILDKITE_PULL_REQUEST}" == "" ]] || [[ "${runAllStages}" == "true" ]] || [[ "${ONLY_DOCS}" == "false" && "${BUILDKITE_PULL_REQUEST}" != "" ]]; then     # from https://github.com/elastic/beats/blob/c5e79a25d05d5bdfa9da4d187fe89523faa42afc/Jenkinsfile#L107-L137
```

Comment:
In my opinion, these conditions are complicated to follow up on; either there are a few nested if branches or a few sequential `or` operators.

Reply:
Totally agree. It is as it is right now. I updated each line of condition with the source of this condition.
It would be great to simplify that. I excluded the conditions from the main pipeline. They will be added during the main pipeline migration.